auth/hash/hash.go

```diff
@@ -0,0 +1,203 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package hash contains a collection of password hash algorithms that can be used with the
+// auth.ImportUsers() API.
+package hash // import "firebase.google.com/go/auth/hash"
+
+import (
+	"encoding/base64"
+	"errors"
+
+	"firebase.google.com/go/internal"
+)
+
+// Bcrypt represents the BCRYPT hash algorithm.
+type Bcrypt struct{}
+
+// Config returns the validated hash configuration.
+func (b *Bcrypt) Config() (*internal.HashConfig, error) {
+	return &internal.HashConfig{HashAlgorithm: "BCRYPT"}, nil
+}
+
+// StandardScrypt represents the standard scrypt hash algorithm.
+type StandardScrypt struct {
+	BlockSize        int
```

Comment:
It seems like the zero values for these params aren't useful. Do we wanna provide any guidance on reasonable defaults?

Reply:
Added links to the public documentation.